keras_core/backend/torch/torch_module_wrapper.py

```diff
@@ -0,0 +1,161 @@
+            )
+            self.flatten = TorchModuleWarpper(nn.Flatten())
+            self.dropout = TorchModuleWarpper(nn.Dropout(p=0.5))
+            self.fc = TorchModuleWarpper(nn.Linear(1600, 10))
+
+        def call(self, inputs):
+            x = F.relu(self.conv1(inputs))
+            x = self.pool(x)
+            x = F.relu(self.conv2(x))
+            x = self.pool(x)
+            x = self.flatten(x)
+            x = self.dropout(x)
+            x = self.fc(x)
+            return F.softmax(x, dim=1)
+
+
+    model = Classifier()
+    model.build((1, 28, 28))
+    print("Output shape:", model(torch.ones(1, 1, 28, 28).to("cuda")).shape)
+
+    model.compile(
+        loss="sparse_categorical_crossentropy",
+        optimizer="adam",
+        metrics=["accuracy"]
+    )
+    model.fit(train_loader, epochs=5)
+    ```
+
+    Here's an example of how the `TorchModuleWarpper` can be used with PyTorch
+    Lazy modules.
+
+    ```python
+    import torch.nn as nn
+    import torch.nn.functional as F
+
+    import keras_core
+    from keras_core.backend.torch import TorchModuleWarpper
+
+
+    class LazyClassifier(keras.Model):
+
+        def __init__(self, *args, **kwargs):
+            super().__init__(*args, **kwargs)
+            # You can wrap all `torch.nn.Module`s with `TorchModuleWarpper`
+            # irrespective of whether they are lazy or not.
+            self.conv1 = TorchModuleWarpper(
+                nn.LazyConv2d(out_channels=32, kernel_size=(3, 3))
+            )
+            self.conv2 = TorchModuleWarpper(
+                nn.LazyConv2d(out_channels=64, kernel_size=(3, 3))
+            )
+            self.pool = TorchModuleWarpper(nn.MaxPool2d(kernel_size=(2, 2)))
+            self.flatten = TorchModuleWarpper(nn.Flatten())
+            self.dropout = TorchModuleWarpper(nn.Dropout(p=0.5))
+            self.fc = TorchModuleWarpper(nn.LazyLinear(10))
+
+        def call(self, inputs):
+            x = F.relu(self.conv1(inputs))
+            x = self.pool(x)
+            x = F.relu(self.conv2(x))
+            x = self.pool(x)
+            x = self.flatten(x)
+            x = self.dropout(x)
+            x = self.fc(x)
+            return F.softmax(x, dim=1)
+
+
+    model = Classifier()
+    model.build((1, 28, 28))
+    print("Output shape:", model(torch.ones(1, 1, 28, 28).to("cuda")).shape)
+
+    model.compile(
+        loss="sparse_categorical_crossentropy",
+        optimizer="adam",
+        metrics=["accuracy"]
+    )
+    model.fit(train_loader, epochs=5)
+    ```
+
+    """
+
+    def __init__(self, module, name=None):
+        super().__init__(name=name)
+        self.module = module.to("cuda")
+        self.lazy = isinstance(self.module, nn.modules.lazy.LazyModuleMixin)
+        if not self.lazy:
+            self.track_module_parameters()
+
+    def parameters(self, recurse=True):
+        return self.module.parameters(recurse=recurse)
+
+    def track_module_parameters(self):
```

Comment:
Just do an in line import of Variable here

Reply:
The inline import of `Variable` doesn't seem to be solving the import issue.
https://github.com/keras-team/keras-core/actions/runs/5985127499/job/16236918699?pr=779

Reply:
Yeah I think this comes from importing `Layer` inside `backend`, which is clearly a circular dependency. Here's what we can do: create `keras_core/utils/torch_utils.py` and move the TorchModuleWrapper there. We already have tf_utils.